sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2204,11 +2204,12 @@ class Analyzer(override val catalogManager: CatalogManager)
         findMethod(scalarFunc, MAGIC_METHOD_NAME, argClasses) match {
           case Some(m) if Modifier.isStatic(m.getModifiers) =>
             StaticInvoke(scalarFunc.getClass, scalarFunc.resultType(),
-              MAGIC_METHOD_NAME, arguments, returnNullable = scalarFunc.isResultNullable)
+              MAGIC_METHOD_NAME, arguments, propagateNull = false,
```

Comment:
how about primitive type parameters?

Reply:
I think it shouldn't matter. Here we're making sure that the magic method will always be invoked **regardless** of null-ness of the arguments. 

For primitive types this has less significant meaning because even if `progagateNull` is true, `needNullCheck`
```
  protected lazy val needNullCheck: Boolean = propagateNull && arguments.exists(_.nullable)
```
in `InvokeLike` also considers null-ness of arguments.

Reply:
How can users handle nullable int values if the UDF is something like `isPositive(int i)` which can't accept null argument?

Reply:
Ah I see what you mean, thanks. Seems we should allow users to define magic method with boxed primitive types for this case? We could also follow the behavior of [ScalaUDF](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala#L32) and returns null if any of the primitive type parameter is nullable and the input is null, however currently `InvokeLike` cannot handle the case where a subset of the input types are of primitive nullable type.